ext/standard/streamsfuncs.c

```diff
@@ -616,93 +616,147 @@ PHP_FUNCTION(stream_get_wrappers)
-{
-	zval *elem;
-	php_stream *stream;
-	int cnt = 0;
+/* {{{ fd_bigset related macros and typedefs to remove FD_SETSIZE limitation in stream_select */
+typedef struct {
+#ifdef __USE_XOPEN
+    char *fds_bits;
+#else
+    char *__fds_bits;
+#endif
+    size_t size;
+} fd_bigset;
+#ifdef __USE_XOPEN
+#define FD_BIGSET_ZERO(set, num_fds) do { \
+    (set)->size = (num_fds + 7) / 8; \
+    (set)->fds_bits = (char *)ecalloc((set)->size, sizeof(char)); \
+} while (0)
+#define FD_BIGSET_SET(fd, set) do { \
+    if ((fd) / 8 < (set)->size) { \
+        (set)->fds_bits[(fd) / 8] |= (1 << ((fd) % 8)); \
+    } \
+} while (0)
+#define FD_BIGSET_ISSET(fd, set) (((fd) / 8 < (set)->size) ? ((set)->fds_bits[(fd) / 8] & (1 << ((fd) % 8))) : 0)
+#define FD_BIGSET_CLR(fd, set) do { \
+    if ((fd) / 8 < (set)->size) { \
+        (set)->fds_bits[(fd) / 8] &= ~(1 << ((fd) % 8)); \
+    } \
+} while (0)
+#define FD_BIGSET_FREE(set) do { \
+    if ((set)->fds_bits) { \
+        efree((set)->fds_bits); \
+        (set)->fds_bits = NULL; \
+    } \
+    (set)->size = 0; \
+} while (0)
+#else
+#define FD_BIGSET_ZERO(set, num_fds) do { \
+    (set)->size = (num_fds + 7) / 8; \
+    (set)->__fds_bits = (char *)ecalloc((set)->size, sizeof(char)); \
+} while (0)
+#define FD_BIGSET_SET(fd, set) do { \
+    if ((fd) / 8 < (set)->size) { \
+        (set)->__fds_bits[(fd) / 8] |= (1 << ((fd) % 8)); \
+    } \
+} while (0)
+#define FD_BIGSET_ISSET(fd, set) (((fd) / 8 < (set)->size) ? ((set)->__fds_bits[(fd) / 8] & (1 << ((fd) % 8))) : 0)
+#define FD_BIGSET_CLR(fd, set) do { \
+    if ((fd) / 8 < (set)->size) { \
+        (set)->__fds_bits[(fd) / 8] &= ~(1 << ((fd) % 8)); \
+    } \
+} while (0)
+#define FD_BIGSET_FREE(set) do { \
+    if ((set)->__fds_bits) { \
+        efree((set)->__fds_bits); \
+        (set)->__fds_bits = NULL; \
+    } \
+    (set)->size = 0; \
+} while (0)
+#endif
 
-	if (Z_TYPE_P(stream_array) != IS_ARRAY) {
-		return 0;
-	}
 
-	ZEND_HASH_FOREACH_VAL(Z_ARRVAL_P(stream_array), elem) {
+int stream_array_to_fd_bigset(zval *stream_array, fd_bigset *set, php_socket_t *max_fd, long max_fds) {
+    zval *elem;
+    php_stream *stream;
+    int cnt = 0;
+
+    if (Z_TYPE_P(stream_array) != IS_ARRAY) {
+        return 0;
+    }
+
+    ZEND_HASH_FOREACH_VAL(Z_ARRVAL_P(stream_array), elem) {
 		/* Temporary int fd is needed for the STREAM data type on windows, passing this_fd directly to php_stream_cast()
 			would eventually bring a wrong result on x64. php_stream_cast() casts to int internally, and this will leave
 			the higher bits of a SOCKET variable uninitialized on systems with little endian. */
 		php_socket_t this_fd;
 
-		ZVAL_DEREF(elem);
-		php_stream_from_zval_no_verify(stream, elem);
-		if (stream == NULL) {
-			continue;
-		}
+        ZVAL_DEREF(elem);
+        php_stream_from_zval_no_verify(stream, elem);
+        if (stream == NULL) {
+            continue;
+        }
```

Comment:
This and other places look like the indentation might be wrong? You can see it in GH if you increase the tab-size to something ridiculous: https://github.com/php/php-src/pull/14452/files?ts=12 and I notice (as I think @ramsey pointed out as well) that you are using spaces instead of tabs to indent.

Reply:
I will fix indentation on the next commit; just want more feedback before I do it.

Reply:
It would be great to do it sooner than later as it's hard to review in the current form.